Documentation/cilium_operator.rst

```diff
@@ -0,0 +1,212 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+Cilium Operator
+===============
+
+.. note:: This document provides a technical overview of the Cilium
+          operator and describes the cluster-wide operations it is
+          responsible for.
+
+Highly Available Cilium Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Cilium operator uses Kubernetes leader election library in conjunction with
+lease locks to provide a HA cluster of cilium-operator instances. The
+capability is supported on Kubernetes versions 1.14 and above and is
+Cilium's default behavior since the 1.9 release.
+
+The number of replicas for the HA deployment can be configured using
+Helm option ``operator.replicas``.
+
+.. parsed-literal::
+
+    helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set operator.replicas=3
+
+.. code:: bash
+
+    $ kubectl get deployment cilium-operator -n kube-system
+    NAME              READY   UP-TO-DATE   AVAILABLE   AGE
+    cilium-operator   3/3     3            3           46s
+
+The operator is an integral part of Cilium installations in Kubernetes
+environments and is tasked to perform the following operations:
+
+CRD Registration
+~~~~~~~~~~~~~~~~
+
+The default behavior of cilium-operator is to register the CRDs used by
+Cilium. The following custom resources are registered by cilium-operator:
+
+-  CiliumNetworkPolicy
+-  CiliumClusterwideNetworkPolicy
+-  CiliumEndpoint
+-  CiliumNode
+-  CiliumExternalWorkload
+-  CiliumIdentity
+-  CiliumLocalRedirectPolicy
+
+IPAM
+~~~~
+
+Cilium-Operator is responsible for IP address management when running in
```

Comment:
Nit: we should either refer this as either `cilium-operator` or Cilium Operator or Operator, and be consistent with it

```suggestion
Cilium Operator is responsible for IP address management when running in
```